gapis/api/transform2/file_log.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transform2
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/config"
+)
+
+type fileLog struct {
+	file *os.File
+}
+
+// NewFileLog returns a Transformer that will log all commands passed through it
+// to the text file at path.
+func NewFileLog(ctx context.Context, path string) *fileLog {
+	f, err := os.Create(path)
+	if err != nil {
+		log.E(ctx, "Failed to create replay log file %v: %v", path, err)
+		return nil
+	}
+	return &fileLog{file: f}
+}
+
+func (logTransform *fileLog) RequiresAccurateState() bool {
+	return false
+}
+
+func (logTransform *fileLog) BeginTransform(ctx context.Context, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
```

Comment:
Variable name longer than type name is not really canonical Go.
Also on naming, your type is not really a fileLog, it is a transform, so fileLogTransform would be more correct.